src/mudlet-lua/lua/geyser/GeyserLabel.lua

```diff
@@ -95,6 +99,15 @@ function Geyser.Label:processFormatString(format)
   end
 end
 
+--- Sets the font face for the label, use empty string to clear the font and use css/default
+-- @param font font face to use
+function Geyser.Label:setFont(font)
+  local af = getAvailableFonts()
+  assert(af[font] or font == "", "attempt to call setFont with font '" .. font .. "' which is not available, see getAvailableFonts() for valid options")
```

Comment:
Fonts as you know aren't consistent across OS's, someone could make a package on windows that'll error on macOS. Best to let Qt decide what fallback font to pick here.

Reply:
It won't set the font to anything except "" if it isn't available in getAvailableFonts(). If the font is set to "" then it doesn't add the font tag to the echo, and uses whatever Qt uses for the fallback

Reply:
Which is to say it won't let you set an invalid font, and if no font is set then the current behaviour is preserved

Reply:
Unless you're saying we should just let it blindly bull forward and let you set fonts which aren't available, and instead of dealing with the "why did this error?" questions, dealing with the "why is my font the same as default after using setFont without error"  questions?

Reply:
So there's two places where it can go wrong - at the development time when the author is making the thing, and at sharing time when someone else is installing it.

We'll definitely cover the former if we make it an error and allow it to work on the developers machine easier, but we'll make the latter case much harder. A person who only ever used Windows isn't going to think twice about including Segoe UI in their package, and when a Mac/Linux user installs it, they'll get a Terminator-like because not blew up on a font they didn't have.

So I think it's best to improve the documentation around setFont to explain "if it didn't change, check the available fonts" and keep the API as non-fragile as possible - so users on other platforms don't get an additional impediment to UIs working out of the box for them.

It'll look different, and thanks to our docs people will figure out the font is missing, but at least it won't be fragile and break.

Reply:
Swallowing errors is an anti pattern. I'll compromise though and change it from assert to if+debugc in the morning so that there is at least something in the error console to tell the user what happened instead of silently behaving unexpectedly. Will also keep from inserting useless font tags or leaving properties set to invalid values.

Reply:
The user shouldn't have to leave the application to be told what went wrong is also part of it.

Reply:
They won't be leaving the application?

Reply:
If the information on why their font isn't changing is in the documentation and not in the error console, then they do have to leave the application for that information. Versus an error or debugc, which puts the information in the error console.

Reply:
That is certainly true, but like I said I'm thinking longer term for all the users down the line - they'll have a better experience this way if things "don't look quite right" vs "this thing is broken because it aborted loading halfway".